### PR TITLE
Firewall nova plugin finished

### PR DESCRIPTION
**Pull request by gmiu**
New files: 
- firewall.py
- sample_firewall.yaml

### CODE REVIEW

**Comment by gmiu** [comment]
@cedwards, @basepi, I'm leaving this here for feedback.

**Comment by basepi** [comment]
This looks great to me!

**Comment by cedwards** [comment]
I'm finding that even blacklisted rules are showing up under failure. how are you validating tests on your end?

**Comment by gmiu** [comment]
Auch, I forgot about blacklist vs whitelist in the `audit` function.

**Comment by cedwards** [comment]
ok. I'll let you sort out that logic and I'll try tests again.

**Comment by gmiu** [comment]
It should be good now (sorry for the wrong commit message)

File: hubblestack_nova/firewall.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                salt_ret = __salt__['iptables.check'](table=table, chain=chain, rule=rule, family=family)
+
+                if salt_ret not in (True, False):
+                    log.error(salt_ret)
+                    passed = False
+                else:
+                    passed = salt_ret
+
+                if tag_data['type'] == 'blacklist':
+                    passed = ~passed
```
Comment (basepi) [inline hubblestack_nova/firewall.py:83]: Can you change this to `passed = not passed` please? It's more pythonic.

**Pull request merged**